File: C++/013_Roman to Integer.cpp
class Solution {
public:
    int romanToInt(string s) {
        int result = 0;
        for (int i = 0; i < s.length(); i++)
            result += get_num(s[i]);
        int pre = 0;
        int cur = get_num(s[0]);
        for (int i = 1; i < s.length(); i++) {
            pre = cur;
            cur = get_num(s[i]);
            if (pre * 5 == cur || pre * 10 == cur)
                result -= pre << 1;
        }
        return result;
    }
private:
    int get_num(char c) {
        switch(c) {
            case 'I': return 1;
            case 'V': return 5;
            case 'X': return 10;
            case 'L': return 50;
            case 'C': return 100;
            case 'D': return 500;
            case 'M': return 1000;
            default: return 0;
        }
    }
};

File: Python/013_Roman to Integer.py
class Solution:
    # @param {string} s
    # @return {integer}
    ctod = {'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000}
    def romanToInt(self, s):
        res = 0
        for i in s:
            res += self._getNum(i)
        pre = 0
        cur = self._getNum(s[0])
        for i in s[1:]:
            pre = cur
            cur = self._getNum(i)
            if pre * 5 == cur or pre * 10 == cur:
                res -= pre << 1
        return res

    def _getNum(self, c):
        return self.ctod.get(c, 0)